### PR TITLE
Add Updated Edition for O'Reilly Idiomatic Go

### PR DESCRIPTION
**Pull request by vrv501**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -260,7 +260,7 @@ In Shipping Go you will learn how to:
  - Scale your deployment in a cost-effective way
  - Deliver a culture of continuous improvement
 
-### 2024 - [Learning Go: An Idiomatic Approach to Real-World Go Programming, 2nd Edition](https://amzn.eu/d/0ExgXvJ)
+### 2024 - [Learning Go: An Idiomatic Approach to Real-World Go Programming, 2nd Edition](https://www.amazon.de/-/en/Jon-Bodner/dp/1098139291)
```
Comment (nathany) [inline README.md:263]: It's probably best to use the publisher (O'Reilly) page instead of Amazon for a specific country.

Comment (nathany) [reply README.md:263]: Are there any updates to the summary in the second edition?

**Comment by nathany** [comment]
Thanks

File: README.md (hunk 1 above)
Comment (vrv501) [reply README.md:263]: The old amazon link was basically same except the id which I felt is standard enforced by the  maintainers. Shall I update it with Oreilly link?

Comment (nathany) [reply README.md:263]: 🤷🏼‍♂️

**Review by nathany — APPROVED** [review]

**Pull request merged**